File: Sort_Integers_by_The_Number_of_1_Bits/sort_1_bits.cpp
class Solution {
public:
    vector<int> sortByBits(vector<int>& arr) {
        sort(arr.begin(),arr.end(), [](int a, int b){
            if(countone(a)!=countone(b)){
                return countone(a)<countone(b);
            }
            else{
                return a<b;
            }
        });
        return arr;

    }
    static int countone(int n){
        int count=0;
        while(n>0){
            count+=n%2;
            n=n/2;
        }
        return count;
    }
};

File: 1356_Sort_Integers_by_The_Number_of_1_Bits/sort_1_Bits.py
class Solution:
    def sortByBits(self, arr: List[int]) -> List[int]:
        return sorted(arr, key=lambda x: (count_ones(x), x))

def count_ones(n):
    count = 0
    while n > 0:
        count += n % 2
        n = n // 2
    return count